graphbolt/src/cnumpy.cc

```diff
@@ -137,86 +156,141 @@ torch::Tensor OnDiskNpyArray::IndexSelectIOUringImpl(torch::Tensor index) {
   auto result_buffer = reinterpret_cast<char *>(result.data_ptr());
 
   // Indicator for index error.
-  std::atomic<bool> error_flag{};
+  std::atomic<int> error_flag{};
   std::atomic<int64_t> work_queue{};
   torch::parallel_for(0, num_thread_, 1, [&](int64_t begin, int64_t end) {
+    if (begin >= end) return;
     const auto thread_id = begin;
-    // Record the inside offsets of fetched features.
-    int64_t residual[kGroupSize];
-    while (true) {
-      begin = work_queue.fetch_add(kGroupSize, std::memory_order_relaxed);
-      if (begin >= index.numel() || error_flag.load()) break;
-      end = std::min(begin + kGroupSize, index.numel());
+    auto &my_io_uring_queue = io_uring_queue_[thread_id];
```

Comment:
my? Why it is not yours :)

Reply:
The class one is `io_uring_queue_`. I thought I could call it `io_uring_queue` but it is too similar to the class member. When thinking of a name, I impersonated the threads we create in the constructor and thought this queue belongs to the thread. Thus, `my_` prefix :).